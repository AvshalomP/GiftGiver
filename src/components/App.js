import React, { Component } from 'react';
import styles from '../styles/App.module.css';
import Gift from './Gift';
import { maxNumber } from '../helper';


class App extends Component {
    constructor(props){
        super(props);

        this.state = { gifts: [] }
    }

    handleAddGift = () => {
        const { gifts } = this.state;
        const ids = gifts.map( gift => gift.id);

        gifts.push({ id: maxNumber(ids)+1});
        this.setState({ gifts });
    };

    removeGift = (id) => {
        const gifts = this.state.gifts.filter( gift => gift.id!==id );

        this.setState({ gifts });
    };

    render(){
        const giftsList = this.state.gifts.map( gift => {
           return ( <Gift key={gift.id} gift={gift} handleRemove={this.removeGift}/> )
        });

        return(
            <div className={styles.container}>
                <h1>Gift Giver</h1>
                <div className="gift-list">
                    {giftsList}
                </div>
                <a id="addBtn" href="#addBtn" className={styles.btn} onClick={this.handleAddGift}>Add Gift</a>
            </div>
        )
    }
}

export default App;