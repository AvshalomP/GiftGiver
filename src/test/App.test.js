import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';


describe('App', () => {
    const app = shallow(<App />);

    it('renders correctly', () => {
        expect(app).toMatchSnapshot();
    });

    it('initializes the `state` with an empty list of gifts', () => {
       expect(app.state().gifts).toEqual([]);
    });

    describe('when clicking the `add gift` button', () => {
        it('adds a new gift to `state`', () => {
            app.find('.btn-add').simulate('click');

            expect(app.state().gifts).toEqual([{ id: 1 }]);
        });
    });
});