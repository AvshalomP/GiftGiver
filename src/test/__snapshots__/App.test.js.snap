// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Gift Giver
        </h1>,
        <div
          className="gift-list"
        />,
        <a
          className="btn"
          href="#addBtn"
          id="addBtn"
          onClick={[Function]}
        >
          Add Gift
        </a>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Gift Giver",
        },
        "ref": null,
        "rendered": "Gift Giver",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "gift-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Gift",
          "className": "btn",
          "href": "#addBtn",
          "id": "addBtn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Gift",
        "type": "a",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Gift Giver
          </h1>,
          <div
            className="gift-list"
          />,
          <a
            className="btn"
            href="#addBtn"
            id="addBtn"
            onClick={[Function]}
          >
            Add Gift
          </a>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gift Giver",
          },
          "ref": null,
          "rendered": "Gift Giver",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "gift-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Gift",
            "className": "btn",
            "href": "#addBtn",
            "id": "addBtn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Gift",
          "type": "a",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
